src/io/encoding.cr

```diff
@@ -216,7 +216,7 @@ class IO
       end
     end
 
-    private def gets_index(index, delimiter, chomp)
+    private def gets_index(index : Int, delimiter, chomp)
```

Comment:
This kind of restriction has a different meaning with the released compiler and compiled compiler, you choose to use Int directly instead of DefaultInt since it was a private method, right?

Reply:
I think I put it because I got an error inside this call and I wanted the error to be outside the call. In general all methods should be restricted to `Int`: "any" `Int` was allowed before, but now I want to only allow the single type `Int`.